src/inference/AdvancedSMC.jl

```diff
@@ -56,35 +95,22 @@ function step(model, spl::Sampler{<:SMC}, vi::VarInfo)
     ## pick a particle to be retained.
     Ws, _ = weights(particles)
     indx = randcat(Ws)
-    push!(spl.info[:logevidence], particles.logE)
-
-    return particles[indx].vi, true
-end
+    push!(spl.state.logevidence, particles.logE)
 
-## wrapper for smc: run the sampler, collect results.
-function sample(model::Model, alg::SMC)
-    spl = Sampler(alg)
+    params = particles[indx].vi[spl]
+    vi_draw = particles[indx].vi
```

Comment:
You need `spl.state.vi = vi_draw` here IIUC.

Reply:
For both PG and SMC I believe.

Reply:
Also worth checking if other samplers make new `VarInfo`s inside `step!` and doing the same if that's the case.

Reply:
@cpfiffer is this resolved?

Reply:
Yes, this has been fixed.